src/main/java/fi/helsinki/cs/tmc/core/domain/Exercise.java

```diff
@@ -272,6 +274,26 @@ public void setAdaptive(boolean adaptive) {
         isAdaptive = adaptive;
     }
 
+    public int getWeek() {
+        return week;
+    }
+
+    public void setWeek(int week) {
+        this.week = week;
+    }
+
+    public void generateWeek() {
+        String weekWord = "viikko";
```

Comment:
While that's slightly more generic and pretty close to what the server is doing right now (see: [exercise.rb#L116-128](https://github.com/testmycode/tmc-server/blob/master/app/models/exercise.rb#L116-L128)), I think that we should consider migrating out of these implicit name-based assumptions in the future. 

That is, I suggest that the server's db schema be updated to include a field `exercise_group` and that the contents of that field should then be used for this kinds of decision everywhere. This, however, is a somewhat involved undertaking as it requires changes to virtually all the TMC components from the server down. I'd say it's outside the scope of this project.